### ISSUE TITLE
Trouble parsing defn forms in datascript

### ISSUE THREAD

**Issue by borkdude** [borkdude/clj-kondo#63]
Not all defns forms in https://github.com/tonsky/datascript/blob/master/src/datascript/db.cljc can be parsed by clj-kondo.

**Comment by borkdude** [comment]
Repro:

```
(ns datascript.db
  (:refer-clojure :exclude [seqable?]))

(defn #?@(:clj  [^Boolean seqable?]
          :cljs [^boolean seqable?])
  [x] x)

(seqable? 1 2)
```

**Comment by borkdude** [comment]
Fixed with 9ef7f0bf2dbeb152b36d86f241ab6c753b5edcb0
